### PR TITLE
Improve resilience of MacOS Code Signing Step

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes Issue #596 

Implements a check for each file signed in the code signing step, validates the results, and has 20 attempts with a minor pause to allow and reduce flooding the code signing service with requests.

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
run tests

**Review by andrew-m-leonard — APPROVED** [review]
lgtm

**Review by adamfarley — APPROVED** [review]
LGTM, well done Scott. :)

**Comment by adamfarley** [comment]
The Windows jdk20 job has failed [here](https://ci.adoptium.net/job/build-scripts-pr-tester/job/build-test/job/jobs/job/jdk20u/job/jdk20u-windows-x64-temurin/14/console), but the error (pasted below) appears identical to one seen 2 weeks ago [here](https://github.com/adoptium/ci-jenkins-pipelines/pull/738#issuecomment-1604448730), so even if this wasn't on a totally different platform, I don't think this failure is related to this PR.

```
11:43:28  fatal: packfile .git/objects/pack/pack-251e1dcc24630754c55ba307bc330d3298caf56a.pack cannot be mapped: Resource temporarily unavailable
11:43:28  fatal: fetch-pack: invalid index-pack output
```

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -1492,10 +1493,45 @@ class Build {
                                                     file=$(basename "$f")
                                                     mv "$f" "${dir}/unsigned_${file}"
                                                     curl -o "$f" -F file="@${dir}/unsigned_${file}" -F entitlements="@$ENTITLEMENTS" https://cbi.eclipse.org/macos/codesign/sign
-                                                    chmod --reference="${dir}/unsigned_${file}" "$f"
-                                                    rm -rf "${dir}/unsigned_${file}"
+                                                    TESTMACSIGN=`grep -i "Apple Certification Authority" "$f"|wc -l`
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1496]: Is there enough logic here to extract into its own function?

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -1492,10 +1493,45 @@ class Build {
...
+                                                          chmod --reference="${dir}/unsigned_${file}" "$f"
+                                                          rm -rf "${dir}/unsigned_${file}"
+                                                          break
+                                                        else
+                                                          echo "$f Failed Signing On Attempt $iteration"
+                                                          iteration=$((iteration+1))
+                                                        fi
+                                                        if [ $iteration -eq $max_iterations ]
+                                                        then
+                                                          echo "Reached Max Attempts"
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1523]: maybe echo what that max is

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 3]
```diff
@@ -1492,10 +1493,45 @@ class Build {
...
+                                                      rm -rf "${dir}/unsigned_${file}"
+                                                    else
+                                                      max_iterations=20
+                                                      iteration=1
+                                                      while [ $iteration -le $max_iterations ]
+                                                      do
+                                                        echo "Code Not Signed - Have Another Try"
+                                                        sleep 1
+                                                        curl -o "$f" -F file="@${dir}/unsigned_${file}" -F entitlements="@$ENTITLEMENTS" https://cbi.eclipse.org/macos/codesign/sign
+                                                        TESTMACSIGN2=`grep -i "Apple Certification Authority" "$f"|wc -l`
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1510]: could extract that grep command into a const?

**Review by karianna — CHANGES_REQUESTED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (sophia-guo) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1496]: What about using flag`--fail`, which will prevent curl from outputting the html file and return error 22.

`curl --fail --silent --show-error -o "$f" -F file="@${dir}/unsigned_${file}" -F entitlements="@$ENTITLEMENTS" https://cbi.eclipse.org/macos/codesign/sign`, similar to https://github.com/adoptium/temurin-build/blob/master/sign.sh#L143

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 2 above)
Comment (steelhead31) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1523]: Good idea.

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 3 above)
Comment (steelhead31) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1511]: Done!

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (steelhead31) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: I think in this context, a shell being run from within a groovy pipeline, this is probably the least impactful solution, especially as its limited to the one job/platform, ie MacOS signing.

**Review by karianna — APPROVED** [review]

**Pull request merged**